### ISSUE TITLE
trigreduce gives malformed property list error

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2647]
Imported from SourceForge on 2024-07-06 17:09:53
Created by **[bigfooted](https://sourceforge.net/u/bigfooted/)** on 2018-02-11 10:22:33
Original: https://sourceforge.net/p/maxima/bugs/3399

---

I try to simplify a horrible trig expression:
~~~
expr:-(2*(B*atan(((A*b+A*a-B)*(cos(B*x)*sin(A*y)+sin(B*x)*cos(A*y)))/(sqrt(A^2*b^2-2*A*B*b-A^2*a^2+B^2)*(sin(B*x)*sin(A*y)-cos(B*x)*cos(A*y)-1)))
    +sqrt(A^2*b^2-2*A*B*b-A^2*a^2+B^2)*atan((cos(B*x)*sin(A*y)+sin(B*x)*cos(A*y))/(sin(B*x)*sin(A*y)-cos(B*x)*cos(A*y)-1))))
 /(A*B*sqrt(A^2*b^2-2*A*B*b-A^2*a^2+B^2));
~~~
 
 So, I try trigreduce:
~~~
 trigreduce(expr);
~~~
 
 malformed property list: (|$b|
                           ((MEXPT SIMP)
                            ((MPLUS SIMP)
                             ((MEXPT SIMP)
                              ((MPLUS SIMP) ((MEXPT SIMP) |$b| 2)
                               ((MTIMES SIMP) -1 ((MEXPT SIMP) |$a| 2)
                                ((MEXPT SIMP) $A 2))
                               ((MTIMES SIMP) -2 |$a| |$b| $B)
                               ((MTIMES SIMP) ((MEXPT SIMP) |$a| 2)
                                ((MEXPT SIMP) $B 2)))
                              ((RAT SIMP) 1 2))
                             ((MTIMES SIMP)
                              ((MEXPT SIMP)
                               ((MPLUS SIMP) ((MEXPT SIMP) |$b| 2)
                                ((MTIMES SIMP) -1 ((MEXPT SIMP) |$a| 2)
                                 ((MEXPT SIMP) $A 2))
                                ((MTIMES SIMP) -2 |$a| |$b| $B)
                                ((MTIMES SIMP) ((MEXPT SIMP) |$a| 2)
                                 ((MEXPT SIMP) $B 2)))
                               ((RAT SIMP) 1 2))
                              ((%COS SIMP)
                               ((MPLUS SIMP) ((MTIMES SIMP) |$b| $X)
                                ((MTIMES SIMP) |$a| $Y)))))
                            -1)
                           ((%SIN SIMP)
                            ((MPLUS SIMP) ((MTIMES SIMP) |$b| $X)
                             ((MTIMES SIMP) |$a| $Y)))).
                             
trigsimp and trigexpand keep the expression unchanged.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:09:55
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-03-05 18:58:22
Original: https://sourceforge.net/p/maxima/bugs/3399/#6df2

---

I think this is the same error:
~~~~
(%i1) trigreduce(atan(tan(x + y + z)));

Maxima encountered a Lisp error:

 malformed property list: ($X $Y $Z).
~~~~
I'm working on a bug fix, I'll keep you posted.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:09:58
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-03-05 19:23:26
Original: https://sourceforge.net/p/maxima/bugs/3399/#04e2

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:10:02
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-03-05 19:23:27
Original: https://sourceforge.net/p/maxima/bugs/3399/#1dd0

---

Fixed by commit 30ed02e0. Closing this report. Thanks for your help.
